plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/MetastoreUtil.java

```diff
@@ -304,6 +309,9 @@ public static void verifyCanDropColumn(HiveMetastore metastore, String databaseN
         if (table.getDataColumns().size() <= 1) {
             throw new TrinoException(NOT_SUPPORTED, "Cannot drop the only non-partition column in a table");
         }
+        if (!ALLOWED_TO_DROP_TABLE_SERDES.contains(table.getStorage().getStorageFormat().getSerde())) {
```

Comment:
Instead of maintaining a `SET` here - Can we add additional method to `HiveStorageFormat` like 
```
    public boolean supportsColumnDropOperation()
    {
        // Only uncompressed text input format is splittable
        return switch (this) {
            case ORC, PARQUET, ... -> true;
            case JSON,  -> false;
        };
    }